mongo/options/clientoptions.go

```diff
@@ -144,6 +145,20 @@ type IDPInfo struct {
 	RequestScopes []string
 }
 
+// CredentialsProvider is the function type that returns AWS credentials.
+type CredentialsProvider func(context.Context) (Credentials, error)
```

Comment:
We should rename this to `AWSCredentialsProvider` since it seems to be specific to the MONGODB-AWS auth mechanism.